dropwizard-cassandra/src/main/java/smartthings/dw/cassandra/CassandraConfiguration.java

```diff
@@ -34,8 +37,9 @@
 	@NotEmpty
 	private String keyspace;
 
-	@NotEmpty
-	private String validationQuery = "SELECT * FROM system.schema_keyspaces";
+	private String validationQuery;
+
+    protected Boolean validationQueryIdempotence;
```

Comment:
Wondering if this should just be a `boolean` with a default value of `true`?

Based on the logic below, it seems the current default behavior is changed from `false` to `true`, but then on override of validation query, you are requiring this flag to be set.

I guess my question is, what situation are you hoping to enable by allowing `null` value for idempotence?

In the context of a health-check, it seems you'd always want it to be idempotent, however I can understand providing flexibility to toggle that behavior.

Reply:
Yes, you want to always have idempotent health check queries; but if you allow overriding the query as a string then there is no way to validate that the query is idempotent. 

The goal with the logic is to set idempotency to true for the default query, since we have validated that. And in case the query has been overridden, force the developer to think explicitly about idempotency instead possibly shooting themselves in the foot. 

An alternative is not allowing to override the health check query.